### PR TITLE
RDM-6946 Implement Functional Tests for 'CCD User Profile Api :: GET /users' operation	

### PR DESCRIPTION
**Pull request by andrewfolga**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/RDM-6496

### Change description ###
Implement Functional Tests for 'CCD User Profile Api :: GET /users' operation:
RDM-6946 Remove importing of definition from setup logic. Fix current test with ANYTHING_PRESENT. Add new functional tests for 'CCD User Profile Api :: GET /users' operation. Cleanup counting logic in UserProfileOperation. Remove the wrong Swagger ApiResponse code.

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: src/aat/java/uk/gov/hmcts/ccd/userprofile/befta/UserProfileTestAutomationAdapter.java [hunk 1]
```diff
@@ -1,109 +1,13 @@
...
 import io.restassured.RestAssured;
-import io.restassured.builder.RequestSpecBuilder;
-import io.restassured.response.Response;
-import io.restassured.specification.RequestSpecification;
-import uk.gov.hmcts.befta.*;
-import uk.gov.hmcts.befta.data.UserData;
-import uk.gov.hmcts.befta.exception.FunctionalTestException;
+import uk.gov.hmcts.befta.DefaultTestAutomationAdapter;
 
 public class UserProfileTestAutomationAdapter extends DefaultTestAutomationAdapter {
```
Comment (andrewfolga) [inline src/aat/java/uk/gov/hmcts/ccd/userprofile/befta/UserProfileTestAutomationAdapter.java]: @MSancaktutar and I discussed removing of user profile func tests importing dependency as it won't work on all environments (preview vs importing on AAT). We should be putting the user profiles using the same API and then trying to get the profiles back as the actual functional test verification.

Comment (MSancaktutar) [reply src/aat/java/uk/gov/hmcts/ccd/userprofile/befta/UserProfileTestAutomationAdapter.java]: Yes, we should actually push required data to the current application under test.

File: src/aat/resources/features/F-097/S-424.td.json [hunk 2]
```diff
@@ -0,0 +1,36 @@
+{
+	"_guid_": "S-424",
```
Comment (MSancaktutar) [inline src/aat/resources/features/F-097/S-424.td.json]: You could define a base object for these test data objects and reuse common fields from it, like operation name, url, method, product name.

File: src/aat/resources/features/F-097/F-097.feature [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+Feature: CCD User Profile Api :: GET /users
+
+  Background:
+    Given an appropriate test context as detailed in the test data source
+
+  @S-424
+  Scenario: should fetch user profile
+    When a successful call [to create user profiles] as in [Standard_User_Profiles_Creation_Data]
+    And a request is prepared with appropriate values
+    And it is submitted to call the [Retrieve user profiles] operation of [CCD User Profile]
```
Comment (MSancaktutar) [inline src/aat/resources/features/F-097/F-097.feature]: We should mention in a separate line that the request contains a jurisdiction id to more clearly describe the business logic of the scenario,

Comment (andrewfolga) [reply src/aat/resources/features/F-097/F-097.feature]: I does not have to pass jurisdiction id hence the reason I did not include it. See here:
https://github.com/hmcts/ccd-user-profile-api/blob/master/src/main/java/uk/gov/hmcts/ccd/endpoint/userprofile/UserProfileController.java#L64-L66

File: src/aat/resources/features/F-097/F-097.feature [hunk 4]
```diff
@@ -0,0 +1,23 @@
+@F-097
+Feature: CCD User Profile Api :: GET /users
```
Comment (MSancaktutar) [inline src/aat/resources/features/F-097/F-097.feature]: Feature: F-07: Get Users

File: src/aat/resources/features/F-097/S-424.td.json [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+    }
+  },
+	"expectedResponse": {
+    "responseCode": 200,
+    "responseMessage": "OK",
+    "headers" : {
+      "Content-Length" : "[[ANYTHING_PRESENT]]",
+      "Content-Type" : "application/json;charset=UTF-8",
+      "Date" : "[[ANYTHING_PRESENT]]"
+    },
```
Comment (MSancaktutar) [inline src/aat/resources/features/F-097/S-424.td.json]: better extend Common_200_Response

File: src/main/java/uk/gov/hmcts/ccd/domain/service/UserProfileOperation.java [hunk 6]
```diff
@@ -84,8 +84,9 @@ private boolean areDefaultsEqual(final UserProfile userProfileFromRepository,
                                   userProfileFromDefintion.getWorkBasketDefaultState())// checkstyle line break
             && userProfileFromRepository.getJurisdictions()
             .stream()
-            .filter(j -> StringUtils.equals(j.getId(), userProfileFromDefintion.getWorkBasketDefaultJurisdiction()))
-            .count() > 0;
+            .anyMatch(j -> StringUtils.equals(
+                j.getId(),
+                userProfileFromDefintion.getWorkBasketDefaultJurisdiction()));
```
Comment (MSancaktutar) [inline src/main/java/uk/gov/hmcts/ccd/domain/service/UserProfileOperation.java]: What was this change for?

File: src/aat/resources/features/F-097/S-424.td.json (hunk 2 above)
Comment (andrewfolga) [reply src/aat/resources/features/F-097/S-424.td.json]: Done.

File: src/main/java/uk/gov/hmcts/ccd/domain/service/UserProfileOperation.java (hunk 6 above)
Comment (andrewfolga) [reply src/main/java/uk/gov/hmcts/ccd/domain/service/UserProfileOperation.java]: In IntelliJ IDE there is a refactoring tool that automatically helps you detect and simplify complex code. Have a try.

File: src/aat/resources/features/F-097/F-097.feature (hunk 4 above)
Comment (andrewfolga) [reply src/aat/resources/features/F-097/F-097.feature]: Doone

File: src/aat/resources/features/F-097/F-097.feature [hunk 7]
```diff
@@ -0,0 +1,23 @@
...
+
+  Background:
+    Given an appropriate test context as detailed in the test data source
+
+  @S-424
+  Scenario: should fetch user profile
+    When a successful call [to create user profiles] as in [Standard_User_Profiles_Creation_Data]
```
Comment (MSancaktutar) [inline src/aat/resources/features/F-097/F-097.feature]: This should be in Given section

File: src/aat/resources/features/F-097/F-097.feature [hunk 8]
```diff
@@ -0,0 +1,23 @@
...
+    And a request is prepared with appropriate values
```
Comment (MSancaktutar) [inline src/aat/resources/features/F-097/F-097.feature]: This should be under When

File: src/aat/resources/features/F-097/S-424.td.json (hunk 5 above)
Comment (andrewfolga) [reply src/aat/resources/features/F-097/S-424.td.json]: Done

File: src/aat/resources/features/F-097/F-097.feature [hunk 9]
```diff
@@ -0,0 +1,23 @@
+@F-097
+Feature: CCD User Profile Api :: GET /users
+
+  Background:
+    Given an appropriate test context as detailed in the test data source
+
+  @S-424
+  Scenario: should fetch user profile
```
Comment (MSancaktutar) [inline src/aat/resources/features/F-097/F-097.feature]: must fetch all user profiles if no jurisdiction is specified

Comment (MSancaktutar) [reply src/aat/resources/features/F-097/F-097.feature]: ++ another scenario for a jurisdiction id being specified

File: src/aat/resources/features/F-097/S-467.td.json [hunk 10]
```diff
@@ -0,0 +1,25 @@
...
+    }
+  },
+	"expectedResponse": {
+    "responseCode": 200,
+    "responseMessage": "OK",
+    "headers" : {
+      "Content-Length" : "[[ANYTHING_PRESENT]]",
+      "Content-Type" : "application/json;charset=UTF-8",
+      "Date" : "[[ANYTHING_PRESENT]]"
+    },
```
Comment (MSancaktutar) [inline src/aat/resources/features/F-097/S-467.td.json]: could we pls do
"_extends_": "Common_200_Response",
?

File: src/aat/java/uk/gov/hmcts/ccd/userprofile/befta/UserProfileBeftaMain.java [hunk 11]
```diff
@@ -1,11 +1,12 @@
 package uk.gov.hmcts.ccd.userprofile.befta;
 
 import uk.gov.hmcts.befta.BeftaMain;
+import uk.gov.hmcts.befta.DefaultTestAutomationAdapter;
 
 public class UserProfileBeftaMain extends BeftaMain {
 
     public static void main(String[] args) {
-        setTaAdapter(new UserProfileTestAutomationAdapter());
+        setTaAdapter(new DefaultTestAutomationAdapter());
```
Comment (MSancaktutar) [inline src/aat/java/uk/gov/hmcts/ccd/userprofile/befta/UserProfileBeftaMain.java]: You don't need to set this adapter. Can just remove this line.
if you would like to use a custom adapter here is the way to do so:
public class DataStoreBeftaMain {

    public static void main(String[] args) {
        BeftaMain.main(args, new DataStoreTestAutomationAdapter());
    }

}

**Pull request merged**